tensorboard/util/test_util.py

```diff
@@ -280,91 +277,37 @@ def ensure_tb_summary_proto(summary):
-  """
-
-  def decorator(f):
-    if inspect.isclass(f):
-      setup = f.__dict__.get("setUp")
-      if setup is not None:
-        setattr(f, "setUp", decorator(setup))
-
-      for name in _unittest_loader.getTestCaseNames(
-          f, unittest.TestLoader.testMethodPrefix):
-        value = getattr(f, name)
-        if (callable(value) and
-            name.startswith(unittest.TestLoader.testMethodPrefix)):
-          setattr(f, name, decorator(value))
-
-      return f
-
-    def decorated(self, *args, **kwargs):
-      if tf2.enabled():
-        self.skipTest(reason)
-
-      return f(self, *args, **kwargs)
-
-    return decorated
-
-  if func is not None:
-    return decorator(func)
-
-  return decorator
+def _run_conditionally(guard, name, default_reason=None):
+  """Creates a decorator that skips a test when guard returns False.
 
+    Args:
+      guard: A lambda that returns True if a test should be executed.
+      name: A human readable name for the decorator for an error message.
+      default_reason: A string describing why a test should be skipped. If it
+          is None, the decorator will make sure the reason is supplied by the
+          consumer of the decorator. Default is None.
 
-def run_v2_only(func=None):
-  """Execute the decorated test only if running in TensorFlow v2 mode.
-
-  This function is intended to be applied to tests that exercise v2 only
-  functionality. If the test is run in v1 mode it will simply be skipped.
-
-  Args:
-    func: function to be annotated. If `func` is None, this method returns a
-      decorator the can be applied to a function. If `func` is not None this
-      returns the decorator applied to `func`.
-
-  Returns:
-    Returns a decorator that will conditionally skip the decorated test method.
-  """
-
-  def decorator(f):
-    if inspect.isclass(f):
-      setup = f.__dict__.get("setUp")
-      if setup is not None:
-        setattr(f, "setUp", decorator(setup))
-
-      for name in _unittest_loader.getTestCaseNames(
-          f, unittest.TestLoader.testMethodPrefix):
-        value = getattr(f, name)
-        if (callable(value) and
-            name.startswith(unittest.TestLoader.testMethodPrefix)):
-          setattr(f, name, decorator(value))
-
-      return f
-
-    def decorated(self, *args, **kwargs):
-      if not tf2.enabled():
-        self.skipTest("Skipping test for TF v2.")
-
-      return f(self, *args, **kwargs)
-
-    return decorated
+    Raises:
+      ValueError when both reason and default_reason are None.
 
-  if func is not None:
-    return decorator(func)
+    Returns:
+      Returns a decorator.
+    """
 
-  return decorator
\ No newline at end of file
+  def _impl(reason=None):
+    if reason is None:
+      if default_reason is None:
+        raise ValueError('%s requires a reason for skipping.' % name)
+      reason = default_reason
+    return unittest.skipUnless(guard(), reason)
+
+  return _impl
+
+_is_tf2 =  tf.__version__.startswith('2.')
```

Comment:
Can you add a TODO here for this to work with internal TF?  Can assign to me if you want.